catalog/model/extension/module/altapay.php

```diff
@@ -32,4 +32,26 @@ public function saveOrderReconciliationIdentifier($order_id, $reconciliation_ide
             }
         }
     }
+
+    /**
+     * @param int $order_id
+     * @param bool $capture
+     * @param bool $refund
+     * @param bool $void
+     *
+     * @return void
+     */
+    public function updateOrderMeta($order_id, $capture = false, $refund = false, $void = false) {
+        $status_query = "";
+        if ($capture) {
+            $status_query = "capture_status='1'";
+        } elseif ($refund) {
+            $status_query = "refund_status='1'";
+        } elseif ($void) {
+            $status_query = "void_status='1'";
+        }
+        if (!empty($status_query)){
+            $this->db->query("UPDATE " . DB_PREFIX . "altapay_orders SET modified='".$this->db->escape((string)date('Y-m-d H:i:s', time()))."', $status_query WHERE order_id='".(int)$order_id."'");
```

Comment:
Casting to integer, can give you a number and overwrite the order.

For example:

$value = (int)"asdasd";

Then value will be equal to 0. So even though the orderId is not correct, it will overwrite order number 0

Reply:
@emicha I've enclosed the SQL in an integer validation check.

Reply:
@maf-embraceit What do we gain from that? It seems that $order_id will still be casted to 0 when the $order_id is not an integer value.

Reply:
@emicha I had removed the type casting and placed a check to verify if the $order_id is an integer >= 1 then proceed. Else the SQL won't be executed.